src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.FindSymbols.Finders;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+using Microsoft.VisualStudio.LanguageServices.FindUsages;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SymbolSearch
+{
+    internal class RoslynSymbolSearchResult : SymbolSearchResult, IResultWithFileLocation, IResultWithContext, IResultWithNamedProject, IResultWithVSGuids, IResultWithReferenceOrDefinition, IResultWithIcon, IResultWithCustomData /* takes care of containin member, containing type and kind */
+    {
+        private DefinitionItem DefinitionItem { get; set; }
+        private SourceReferenceItem ReferenceItem { get; set; }
```

Comment:
Since this is optional, please mark as nullable. (Also ditto on @CyrusNajmabadi's question of why mutable.)